lib/util/http/index.ts

```diff
@@ -126,20 +127,34 @@ export class Http<GetOptions = HttpOptions, PostOptions = HttpPostOptions> {
         return resolveResponse<T>(cachedRes, options);
       }
     }
-    const startTime = Date.now();
-    const promisedRes = got<T>(url, options);
+    let startTime: number;
+    const queueTask = (): Promise<Response<T>> => {
+      const queueTime = Date.now();
+      const queueDuration = queueTime - startTime;
+      return got<T>(url, options).then((resp) => {
```

Comment:
We have very little remaining `.then()` use in the repo. not possible to structure this as async/await?